editoast/src/models/paced_train.rs

```diff
@@ -181,27 +181,38 @@ impl PacedTrain {
     pub fn iter_occurrences(&self) -> impl Iterator<Item = (TrainId, TrainSchedule)> {
         let mut base_occurrences = self.get_base_occurrences();
 
-        let enabled_modified_exceptions = self
+        let modified_exceptions = self
             .exceptions
             .iter()
-            .filter(|ex| !ex.disabled)
             .filter_map(|e| match e.exception_type {
                 ExceptionType::Modified { occurrence_index } => Some((occurrence_index, e)),
                 _ => None,
             });
 
-        for (occurrence_index, exception) in enabled_modified_exceptions {
+        let mut to_remove = vec![false; base_occurrences.len()];
+        // Modify corresponding occurrences.
+        for (occurrence_index, exception) in modified_exceptions {
             if let Some(occurrence) = base_occurrences.get_mut(occurrence_index as usize) {
-                let train_id = TrainId::PacedTrainModifiedException {
-                    paced_train_id: self.id,
-                    index: occurrence_index as u64,
-                    exception_key: exception.key.clone(),
-                };
-                *occurrence = (train_id, self.apply_exception(exception));
+                if exception.disabled {
+                    to_remove[occurrence_index as usize] = true;
+                } else {
+                    let train_id = TrainId::PacedTrainModifiedException {
+                        paced_train_id: self.id,
+                        index: occurrence_index as u64,
+                        exception_key: exception.key.clone(),
+                    };
+                    *occurrence = (train_id, self.apply_exception(exception));
+                }
             }
         }
+        // Remove disabled occurrences.
+        let occurences = base_occurrences
+            .into_iter()
+            .zip(to_remove)
+            .filter_map(|(occ, disabled)| if disabled { None } else { Some(occ) })
+            .collect_vec();
```

Comment:
We don't need to collect since we're returning an iterator, let's keep things lazy.
```suggestion
            .filter_map(|(occ, disabled)| if disabled { None } else { Some(occ) });
```